### ISSUE TITLE
Disparity in magnitude of router Output

### ISSUE THREAD

**Issue by ZhangYi1999** [huiyiwang01/SEMA-CL#1]
Dear Author of SEMA,

I have recently read about this project, which is very impressive. However, when I try to reproduce it, I found a issue about the design of the expandable router. In this project, the router is initated as one column matrix (in code `nn.Linear(D, 1)`). However, if such router trained on the first task, since the output of router will be normalized by `F.softmax()` and currently only one router, no matter what it outputs will become 1, the first router tends to output a very small value (in my case from -0.5 to 0.4). When training the second router, the router will output a larger number so that more weights will be added on the new adapter. (in my case from 5.0 to 10.0). This brings a huge routing unbalance in class incremental learning.

I wonder if you have encoutered similar situation. If so, have you tried any technique to mitigate it?

Best Regrads,
Yi Zhang

**Comment by yaohui120** [comment]
+1. I also have the problem about how the router avoid catastrophic forgetting.
